datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -22,12 +22,23 @@ use arrow::buffer::Buffer;
 use arrow::compute;
 use arrow::datatypes::{DataType, Field};
 use core::any::type_name;
-use datafusion_common::cast::as_list_array;
+use datafusion_common::cast::{as_generic_string_array, as_int64_array, as_list_array};
 use datafusion_common::ScalarValue;
 use datafusion_common::{DataFusionError, Result};
 use datafusion_expr::ColumnarValue;
 use std::sync::Arc;
 
+macro_rules! downcast_arg {
```

Comment:
I think this is the same as `downcast_value`: https://docs.rs/datafusion-common/26.0.0/datafusion_common/macro.downcast_value.html